### PR TITLE
feat: Add Ruby support

### PR DESCRIPTION
**Pull request by james-francis-MT**
Adding ruby detection to ci-image-builder

For now can only detect ruby version via a Gemfile

### CODE REVIEW

File: test/image_builder/codebase/language/test_ruby.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+from unittest.mock import patch
+
+import pytest
+from parameterized import parameterized
+
+from image_builder.codebase.language import RubyLanguage
+from image_builder.codebase.language.base import CodebaseLanguageNotDetectedError
+
+
+class TestCodebaseLanguagePython(BaseTestCase):
```
Comment (rav-pradhan) [inline test/image_builder/codebase/language/test_ruby.py:14]: Assuming:
```suggestion
class TestCodebaseLanguageRuby(BaseTestCase):
```

Comment (james-francis-MT) [reply test/image_builder/codebase/language/test_ruby.py:14]: Nice, good spot

**Review by rav-pradhan — APPROVED** [review]

**Review by acodeninja — APPROVED** [review]
Looks good to me, though I think it's an idea to get a review from the platform team.

**Review by jackvdp — APPROVED** [review]
Nice!

**Review by yusufsheiqh — APPROVED** [review]

**Comment by WillGibson** [comment]
I think [this part of the README](https://github.com/uktrade/ci-image-builder/blob/main/README.md?plain=1#L3-L4) wants updating to cover what we are supporting.

File: test/image_builder/codebase/language/test_ruby.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+            ('"3.2.1"', "3.2"),
+            ('"3.3.4"', "3.3"),
+            ('"3.3"', "3.3"),
+            ("'3.2.1'", "3.2"),
+            (" '3.2.1'", "3.2"),
+            ('"3.3.x"', "3.3"),
+            ('"3.11.8"', "3.11"),
+        ]
+    )
+    def test_getting_version_from_pyproject(self, input_version, output_version):
```
Comment (WillGibson) [inline test/image_builder/codebase/language/test_ruby.py:34]:
```suggestion
    def test_getting_version_from_gemfile(self, input_version, output_version):
```

**Comment by james-francis-MT** [comment]
> I think [this part of the README](https://github.com/uktrade/ci-image-builder/blob/main/README.md?plain=1#L3-L4) wants updating to cover what we are supporting.

I've added some information about ruby. I'm not sure if anything else needs updating in the readme

**Review by WillGibson — APPROVED** [review]

**Pull request merged**